core/src/main/scala/kafka/tools/ConsoleConsumer.scala

```diff
@@ -100,8 +101,8 @@ object ConsoleConsumer extends Logging {
         consumer.receive()
       } catch {
         case e: Throwable => {
-          error("Error processing message, stopping consumer: ", e)
-          consumer.stop()
+          error("Error processing message, terminating consumer process: ", e)
+          // Consumer will be stopped in shutdown hook
```

Comment:
@ijuma @guozhangwang In the exception case, consumer is not longer inside a poll. It returns and calls `consumer.cleanup()` in the finally block, which closes the consumer (new and old). If `consumer.stop()` is invoked on the new consumer before calling `consumer.cleanup()` when the consumer is not actually polling, `cleanup()` throws `ConsumerWakeupException` and doesn't close cleanly. 

In shutdown hook on the other hand, `consumer.stop()` is required to stop the receive thread if it is inside a poll. For the new consumer, `consumer.cleanup()` may be useful too in the shutdown hook, but it may have been omitted to shutdown quickly.